### PR TITLE
ResourceLoader: Report error if resource type unrecognized (reverted)

### LINKED ISSUE #95490: ResourceLoader.load_threaded_get_status returns THREAD_LOAD_LOADED after ResourceLoader.load_threaded_request with invalid path
### Tested versions

- Reproducible in v4.3.rc3.official [03afb92ef], latest master [9f6336b46], v4.0.4.stable.official [fc0b241c9]

### System information

Godot v4.3.rc3 - Ubuntu 24.04 LTS 24.04 - X11 - GLES3 (Compatibility) - Mesa Intel(R) Graphics (ADL GT2) - 12th Gen Intel(R) Core(TM) i5-1235U (12 Threads)

### Issue description

`ResourceLoader.load_threaded_get_status` with invalid path returns `3`, which is `THREAD_LOAD_LOADED`, after `ResourceLoader.load_threaded_request` with that same invalid path. Interestingly, `ResourceLoader.load_threaded_request` returns `0`, which is `OK`, while printing an error. After call to `ResourceLoader.load_threaded_get` with that invalid path, `ResourceLoader.load_threaded_get_status` starts to give normal result: `0`, which is `THREAD_LOAD_INVALID_RESOURCE`.
By invalid path, i mean something like `res://node.tsc` or invalid UID, NOT like `res://node.gd`, which results in "File not found" and `THREAD_LOAD_FAILED` status.

### Steps to reproduce

1. Call `ResourceLoader.load_threaded_request` with some invalid path.
2. Call `ResourceLoader.load_threaded_get_status` with that same path. Notice status equaling to 3.

### Minimal reproduction project (MRP)

[mrp.zip](https://github.com/user-attachments/files/16601878/mrp.zip)
 1. Start project.
 2. Notice output.
 3. Click "get" to call `ResourceLoader.load_threaded_get`.

### PR DESCRIPTION
**Pull request by RandomShaper**
Fixed version of the reverted #95508.

~Not tested yet, but maybe a brave reviewer can confirm this is good now.~

Fixes #95490.

### CODE REVIEW

**Comment by RandomShaper** [comment]
Cherry-pick for 4.3 included in #96606.

File: core/io/resource_loader.cpp [hunk 1]
```diff
@@ -306,11 +306,16 @@ Ref<Resource> ResourceLoader::_load(const String &p_path, const String &p_origin
 	ERR_FAIL_COND_V_MSG(found, Ref<Resource>(),
 			vformat("Failed loading resource: %s. Make sure resources have been imported by opening the project in the editor at least once.", p_path));
 
+	DEV_ASSERT(!found);
+
 #ifdef TOOLS_ENABLED
 	Ref<FileAccess> file_check = FileAccess::create(FileAccess::ACCESS_RESOURCES);
 	ERR_FAIL_COND_V_MSG(!file_check->file_exists(p_path), Ref<Resource>(), vformat("Resource file not found: %s (expected type: %s)", p_path, p_type_hint));
```
Comment (Summersay415) [inline core/io/resource_loader.cpp:313]:
```suggestion
	if (!file_check->file_exists(p_path)) {
		if (r_error) {
			*r_error = ERR_FILE_NOT_FOUND;
		}
		ERR_FAIL_V_MSG(Ref<Resource>(), vformat("Resource file not found: %s (expected type: %s)", p_path, p_type_hint));
	}
```

**Review by Summersay415 — APPROVED** [review]
Works with suggested change. Fixes #95490 and does not reintroduce #97153.

P. S.: _order matters..._

**Comment by RandomShaper** [comment]
@Summersay415, I've integrated your changes and made further changes so error handling is clearer.

**Comment by clayjohn** [comment]
@RandomShaper Have you tested this now with the latest version?

**Comment by RandomShaper** [comment]
Yes, works as expected.

**Review by clayjohn — APPROVED** [review]
Seems straightforward

**Comment by Repiteo** [comment]
Thanks!

**Comment by akien-mga** [comment]
Reverted by #99469 to fix #99467.

**Pull request merged**